from .db import (
    aggregate,
    count_documents,
    create_index,
    delete_many,
    delete_one,
    ensure_document_exists,
    find_by_id,
    find_many,
    find_one,
    insert_many,
    insert_one,
    update_many,
    update_one,
)
from .exceptions import DatabaseError, DocumentNotFoundError, InvalidObjectIdError
from .helpers import (
    clean_phone_number,
    deep_merge_dicts,
    format_datetime,
    generate_slug,
    paginate_results,
    safe_float,
    safe_int,
    truncate_string,
    validate_email_format,
)
from .security import (
    SecurityError,
    generate_token,
    generate_tokens,
    hash_password,
    sanitize_user_data,
    validate_password_strength,
    verify_password,
)

__all__ = [
    # Exceptions
    "DatabaseError",
    "InvalidObjectIdError",
    "DocumentNotFoundError",
    # Database operations
    "insert_one",
    "insert_many",
    "find_one",
    "find_by_id",
    "find_many",
    "update_one",
    "update_many",
    "delete_one",
    "delete_many",
    "count_documents",
    "aggregate",
    "ensure_document_exists",
    "create_index",
    # Security
    "hash_password",
    "verify_password",
    "generate_tokens",
    "generate_token",
    "validate_password_strength",
    "sanitize_user_data",
    "SecurityError",
    # Helpers
    "paginate_results",
    "validate_email_format",
    "clean_phone_number",
    "format_datetime",
    "safe_int",
    "safe_float",
    "truncate_string",
    "generate_slug",
    "deep_merge_dicts",
]
